### ISSUE TITLE
Backup incomplete, corrupting running database under certain circumstance!

### ISSUE THREAD

**Issue by gerryfrancis** [deltachat/deltachat-android#1811]
**WARNING!! Reproducing this issue by performing the steps mentioned below will likely corrupt your database!**

- Operating System (Linux/Mac/Windows/iOS/Android):
Android 10 on Fairphone FP2 (Lineage OS 17.1 build QQ3A.200805.001 from 2021-02-05).

- Delta Chat Version:
1.15.0-beta (nightly build from 2021-02-16).

- Expected behavior:
A complete backup is performed, without corrupting the current database.

- Actual behavior:
The backup file is corrupted, although sometimes Delta Chat says "successfully stored". In addition, the CURRENT database is corrupted as well!

- Steps to reproduce the problem:
-- Go to Android app settings and end Delta Chat.
Perform the following steps very speedy:
-- Start Delta Chat (with the launcher).
-- From the three-dot menu select "Settings", then "Chats and media".
-- Scroll to the bottom.
-- Tap on "Delete messages from device", then select "Cancel".
-- Tap on "Delete messages from server", then select "Cancel".
-- Tap on "Backup chats to external storage", then select "Start Backup".

- Screenshots:
![grafik](https://user-images.githubusercontent.com/51095533/108216705-0d93e880-7133-11eb-8e2e-00a99d8883d2.png)

- Logs:
[deltachat-log-20210217-135658.txt](https://github.com/deltachat/deltachat-android/files/5996088/deltachat-log-20210217-135658.txt)

- Remarks:
-- Sometimes when the backup fails, a message "IMAP Connect without configured parameters" is displayed on the screen.
-- I consider this issue to be a blocker because the user is not aware that the backup is corrupted. **Furthermore, this procedure seems to corrupt the database currently running, e.g. avatars for contacts are no longer displayed.**

**Comment by Hocuri** [comment]
I could never reproduce this on my test phone (current master, FP3, Stock Android 10).

I coudn't really make sense of the log, either: There are 3000 lines logged in 2s, because DC tries to run housekeeping very, very often, and as soon as it fails, it tries to run it again. (It fails as the database is closed, which is closed as export is currently running). So, one would assume that DC runs into a loop.

But the log lines don't just repeat over and over again, as one would expect in a loop; e.g. sometimes it says "loading job for Imap-thread" before every round of housekeeping, sometimes not. Maybe the log lines are somehow out of order? Or separate threads with housekeeping-loops run at the same time?

@gerryfrancis 
1. how reliably can you reproduce this with the steps above? Does it happen everytime?
2. If you leave out the two "Delete messages from *" steps and instead, wait a little, does it still happen?

**Comment by gerryfrancis** [comment]
> Does it happen everytime?

@Hocuri No, unfortunately not. It has been very difficult for me, too, to reproduce the issue. But corruption occurs as well when a message is received while the backup process is in progress, as I have experienced right now. At least this scenario is reproducible (with a little help from a friend)... ;)

> If you leave out the two "Delete messages from *" steps and instead, wait a little, does it still happen?

@Hocuri No.

**Comment by Hocuri** [comment]
> But corruption occurs as well when a message is received while the backup process is in progress, as I have experienced right now. 

Great, I have an almost-failing test now! almost as everything except for the log seems to be OK, we can just let the test check the log but it's still curious

**Comment by gerryfrancis** [comment]
@Hocuri If you terminate the app in the Android app settings and start over again, you will see that avatars are no longer displayed, which is what I have meant with "corrupted database"... ;)

**Comment by TNCS-git** [comment]
There is also another way to corrupt the database currently running, i.e. avatars for contacts are no longer displayed when the app is newly started.

If you let the DeltaChat to move to the e-mail to DeltaChat folder instead of creating a e-mail host based rule, upon new DeltaChat e-mail message arrival, if your mail provider mark this e-mail spam and DeltaChat was quick enough to catch this e-mail in Inbox before mail provider mark and move the mail to spam (there sometimes is a minor delay for marking Spam action by the host), DeltaChat db will corrupt for that user upon next restart with avatar gone.

Which is why I no longer let DeltaChat watch over any other folder (Inbox or Sent) except just the DeltaChat folder, and I move the e-mail using e-mail server hosting based rule.

**Comment by gerryfrancis** [comment]
@TNCS-git Many thanks for sharing your issue! :)
